### DISCUSSION TITLE
Including an external pyd file

### DISCUSSION THREAD

**Discussion by ughstudios** [pyinstaller/pyinstaller#7604] [Help]
I have a pyd file which is part of my local repository. The executable of the pyinstaller application is committed to source control and pulled on local users workstations. However, I do not want pyinstaller to "copy" the pyd file. The pyd file could be updated independently of the application, but the application requires the pyd file. Is there a way to do this?

**Comment by rokm** [comment]
What exactly is the problem? The pyd extension should be loadable as long as it is placed in location that's in `sys.path`. Same as under regular python.

So for onedir program, you can just manually copy it to the top-level application directory (which is automatically added to `sys.path`). For onefile, you'd need to add location of the pyd file to `sys.path` before trying to import it.

**Reply by ughstudios** [reply]
Should you just use the `--path` argument for adding it to the `sys.path`?

**Reply by rokm** [reply]
`--path` is build-time option and affects search locations during import analysis. If you want to modify run-time search path to include an external location (which I assume you are after, since you said you don't want the pyd to be part of the frozen application), you need to manipulate `sys.path` within your code.
